### ISSUE TITLE
Failed assertion not caught within nested test suite

### ISSUE THREAD

**Issue by sawmurai** [dubzzz/fast-check#6117]
## 🐛 Bug Report

Running this reproducer some tests should actually fail: https://stackblitz.com/edit/vitest-dev-vitest-7pgu3p7a?file=test%2Frepro.test.ts

## To Reproduce

Steps to reproduce:

Open https://stackblitz.com/edit/vitest-dev-vitest-7pgu3p7a?file=test%2Frepro.test.ts and see the running tests in the console.

Here is a copy should the stackblitz noch be available:
```ts
import * as vi from 'vitest';
import { fc, it, test } from '@fast-check/vitest';

vi.describe('outer test that works', () => {
  it.prop([fc.integer()])('reports the seed', (num) => {
    vi.expect(num).toBe(num);
  });
});

vi.describe('outer test that does not work', () => {
  it.prop([fc.integer()])('reports the seed', (num) => {
    test('some inner test', () => {
      // this should actually fail
      vi.expect(1).toBe(0);
    });

    vi.describe('some inner test', () => {
      // this should actually fail
      vi.expect(1).toBe(0);
    });
  });
});
```

## Expected behavior

Tests should fail, but they don't

## Your environment

| Packages / Softwares | Version(s) |
| -------------------- | ---------- |
| fast-check           |     ^4.1.1       |
| node                 |       20.19.1     |
| TypeScript\*         |            |

\*Only for TypeScript's users

**Comment by dubzzz** [comment]
Unfortunately we don't support calling `test` or `it` within `fc.assert` or `it.prop`. It's as is you were calling `test` within an `it` in Vitest or Jest, I'm not sure they will do anything good with it.

**Comment by dubzzz** [comment]
Running `test` in `test` on Vitest results on:

```
 FAIL  test/repro.test.ts > reports the seed
Error: Calling the test function inside another test function is not allowed. Please put it inside "describe" or "suite" so it can be properly collected.
 ❯ null.<anonymous> test/repro.test.ts:5:3
      3| 
      4| test('reports the seed', () => {
      5|   test('some inner test', () => {
```

For the snippet:

```js
import * as vi from 'vitest';
import { test } from 'vitest';

test('reports the seed', () => {
  test('some inner test', () => {
    // this should actually fail
    vi.expect(1).toBe(0);
  });
});
```

**Comment by sawmurai** [comment]
Yeah, its really confusing. Wrapping things into a `describe` seems to "fix" it

```ts
import * as vi from 'vitest';
import { test, describe } from 'vitest';

test('reports the seed', () => {
  describe('this is madness', () => {
    test('some inner test', () => {
      // this should actually fail
      vi.expect(1).toBe(0);
    });
  });
});
``` 

Unfortunately we have quite a deep nesting of this so I was hoping fast-check was meant to support it :)

**Comment by dubzzz** [comment]
Actually in the suggestion https://github.com/dubzzz/fast-check/issues/6117#issuecomment-3210823004, Vitest don't report the test failure of "some inner test".

https://stackblitz.com/edit/vitest-dev-vitest-xhvewer8?file=test%2Frepro.test.ts

**Comment by sawmurai** [comment]
Oh indeed! I checked our codebase and we have tons of `describe` within `describe`, but so far I could not find any `describe` within `it` or `test` which seems to be the boundary. 

But anyway, thank you for taking a look! My question is answered: this is not a bug :)

Feel free to close the issue

**Comment by dubzzz** [comment]
Let me know if I can help with this test. Maybe there is a real issue I have not captured yet from your example
